l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+    }
+
+    function initializeConfig() internal {
+        // Grab config from output of l1 deployment
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/prepare-registration-calldata.toml");
+        string memory toml = vm.readFile(path);
+
+        // Config file must be parsed key by key, otherwise values returned
+        // are parsed alfabetically and not by key.
+        // https://book.getfoundry.sh/cheatcodes/parse-toml
+        config.bridgehub = toml.readAddress("$.deployed_addresses.bridgehub_proxy_addr");
+        config.stateTransitionProxy = toml.readAddress("$.deployed_addresses.state_transition_proxy_addr");
+        config.l1SharedBridgeProxy = toml.readAddress("$.deployed_addresses.l1_shared_bridge_proxy_addr");
+        config.erc20BridgeProxy = toml.readAddress("$.deployed_addresses.erc20_bridge_proxy_addr");
+
+        config.chainId = toml.readUint("$.chain.chain_id");
+        config.eraChainId = toml.readUint("$.chain.era_chain_id");
+        config.chainAdmin = toml.readAddress("$.chain.admin");
+        config.diamondCutData = toml.readBytes("$.chain.diamond_cut_data");
+        config.bridgehubCreateNewChainSalt = toml.readUint("$.chain.bridgehub_create_new_chain_salt");
+        config.baseToken = toml.readAddress("$.chain.base_token_addr");
+        config.governance = toml.readAddress("$.chain.governance_addr");
+
+        bytecodes.l2SharedBridgeBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/contracts/bridge/L2SharedBridge.sol/L2SharedBridge.json"
+        );
+
+        bytecodes.l2SharedBridgeProxyBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol/TransparentUpgradeableProxy.json"
+        );
+        bytecodes.beaconProxy = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol/BeaconProxy.json"
+        );
+    }
+
+    function checkBaseTokenAddress() internal view {
+        if (config.baseToken == address(0)) {
+            revert("Token address is not set");
+        }
+
+        // Check if it's ethereum address
+        if (config.baseToken == ADDRESS_ONE) {
+            return;
+        }
+
+        if (config.baseToken.code.length == 0) {
+            revert("Token address is an EOA");
+        }
+
+        console.log("Using base token address:", config.baseToken);
+    }
+
+    function prepareRegisterBaseTokenCall() internal view returns (IGovernance.Call memory) {
+        Bridgehub bridgehub = Bridgehub(config.bridgehub);
+
+        bytes memory data = abi.encodeCall(bridgehub.addToken, (config.baseToken));
+
+        return IGovernance.Call({target: config.bridgehub, value: 0, data: data});
+    }
+
+    function computeL2BridgeAddress() internal view returns (address) {
```

Comment:
add a comment what it does (as it tries to estimate the address based on the bytecode)